src/lib.rs

```diff
@@ -306,13 +306,25 @@ impl KVMIntrospectable for KVMi {
         Ok(vcpu_count)
     }
 
-    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
+    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, KvmMsr), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let mut msrs: KvmMsr = unsafe { mem::MaybeUninit::<KvmMsr>::zeroed().assume_init() };
```

Comment:
One more thing: Obviously, if you allocate `kvm_msrs` like I suggested you end up with a data structure that is unsafe to use. I would implement an accessor function for `kvm_msrs` that returns a slice (`std::slice::from_raw_parts()`) in order to make it safe to access in libmicrovmi.